### PR TITLE
related to JUNEAU-49 and its inclusion via STREAMS-504

### PR DESCRIPTION
**Pull request by steveblackmon**

### CODE REVIEW

File: juneau-core/src/main/java/org/apache/juneau/internal/StringUtils.java [hunk 1]
```diff
@@ -618,7 +618,9 @@ public static boolean isEmpty(String s) {
 	 * @return <jk>true</jk> if specified string is <jk>null</jk> or it's {@link #toString()} method returns an empty string.
 	 */
 	public static boolean isEmpty(Object s) {
-		return s == null || s.toString().isEmpty();
+		if( s == null ) return true;
+		if( s instanceof List ) return ((List) s).size() == 0;
```
Comment (johnament) [inline juneau-core/src/main/java/org/apache/juneau/internal/StringUtils.java]: Would any `Collection` apply here? Or specifically `List`?

Comment (clr-apache) [inline juneau-core/src/main/java/org/apache/juneau/internal/StringUtils.java]: There is a big discussion on stackoverflow on this topic. 
https://stackoverflow.com/questions/3321526/should-i-use-string-isempty-or-equalsstring
I like the simple 
"".equals(s)
which handles the null case.

If you include s being an instance of collection, it seems that you could use
if (s instanceof Iterable) return (Iterable) s).isEmpty();
Then even user-defined types that implement Iterable can be used.

Comment (clr-apache) [inline juneau-core/src/main/java/org/apache/juneau/internal/StringUtils.java]: Of course, I missed the semantic that you want to return true if s is null. So you need the null check in addition to the equals or isEmpty check.

Too much javascript for me I guess. :(

Comment (johnament) [inline juneau-core/src/main/java/org/apache/juneau/internal/StringUtils.java]: @clr-apache the only problem with that is `Collection` is what introduces `isEmpty()`, `Iterable` only provides a handle to the `Iterator`

Comment (clr-apache) [inline juneau-core/src/main/java/org/apache/juneau/internal/StringUtils.java]: of course you are right.

**Pull request closed without merging**